src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -127,10 +128,18 @@ public static IEnumerable<DType> GetPrimitiveTypes()
         // This is only used by attachment type. DocumentDataType is used to workaround the issue of having cycles in struct type.
         protected readonly DType _attachmentType;
 
-        internal HashSet<IExternalTabularDataSource> AssociatedDataSources { get; private set; }
+        internal HashSet<IExternalTabularDataSource> AssociatedDataSources { get; }
         internal IExternalOptionSet<int> OptionSetInfo { get; }
         internal IExternalViewInfo ViewInfo { get; }
 
+        // Eventually, all display names should come from this centralized source
+        // We should not be using individual DataSource/OptionSet/View references
+        internal DisplayNameProvider DisplayNameProvider { get; private set; }
+
+        // If the type is derived from multiple DisplayNameProviders, disable display names.
+        private bool DisableDisplayNames = false;
```

Comment:
>DisableDisplayNames [](http://example.com/codeflow?start=21&length=19)

Is this the same as setting DisplayNameProvider to null?

Reply:
No, let's say you had 3 different sources that get unioned into a single type. 
Then you'd wind up toggling it back to enabled if we were only going off of nullness of DNP.